Runtime/Scripts/Attributes/RGStateAttribute.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+
+namespace RegressionGames
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Field, Inherited = false, AllowMultiple = false)]
+    public class RGStateAttribute : Attribute
+    {
+        public string StateName { get; }
+
+        // This constructor allows the attribute to be used without a name parameter, e.g., [RGState]
+        public RGStateAttribute()
+        {
+        }
+
+        // This constructor allows a name parameter to be specified, e.g., [RGState("Some Name")]
+        public RGStateAttribute(string stateName)
+        {
+            StateName = stateName;
```

Comment:
This question is likely out of scope for this task, but in the spirit of thinking ahead... based on Aaron's proposal for the bot dashboard design, we know we're going to want additional params that customize how the attribute is displayed in the client (ex. type = "bar", min = "0", max = maxHealth). So question is, can these attributes accept a map or an instance of a "config" object as a parameter? Something that would allow me to specify only the fields that I want to (ie. I don't need to pass a name or type if I really only want to set a min and max value), and won't require us to have many different constructors. From a quick search, it looks like this may not be possible, so we'll probably want to think about alternatives to keep these easy for the user to define and customize